File: SwiftAlgorithm/BOJ/그리디,정렬/1931.swift
//
//  main.swift
//  SwiftAlgorithm
//
//  Created by 유정주 on 2022/03/12.
//

//1931 회의실 배정
import Foundation

let count = Int(readLine()!)!
var times: [(Int, Int)] = []
for _ in 0..<count {
    let input = readLine()!.split(separator: " ").map { Int(String($0))! }
    
    times.append((input[0], input[1]))
}

times.sort { $0.1 == $1.1 ? $0.0 < $1.0 : $0.1 < $1.1 }

var result = 0
var endTime = -1
for time in times {
    if endTime <= time.0 {
        result += 1
        endTime = time.1
    }
}

print(result)

File: Python/BOJ/그리디/1931.py
//
//  1931.py
//  SwiftAlgorithm
//
//  Created by 유정주 on 2023/06/05.
//

import sys; readLine = sys.stdin.readline

n = int(readLine())
items = []
for _ in range(n):
  start, end = map(int, readLine().split())
  items.append((start, end))

items.sort(key= lambda x: (x[1], x[0]))

result = 0
endTime = 0
for item in items:
  if item[0] >= endTime:
    result += 1
    endTime = item[1]

print(result)
